src/MICore/MIResults.cs

```diff
@@ -31,7 +31,7 @@ public enum ResultClass
         exit
     }
 
-    public class ResultValue
+    public abstract class ResultValue
```

Comment:
What's the purpose of all the changes to MIResults.cs? I don't see any motivation behind these changes except style.